shub/config.py

```diff
@@ -164,15 +171,47 @@ def get_version(self):
         elif self.version:
             return str(self.version)
 
+    def get_stack(self, target):
+        return self.stacks.get(target)
+
     def get_target(self, target, auth_required=True):
         """Return (project_id, endpoint, apikey) for given target."""
-        return (
-            self.get_project_id(target),
-            self.get_endpoint(target),
-            self.get_apikey(target, required=auth_required),
+        return Target(
+            name=target,
+            project_id=self.get_project_id(target),
+            endpoint=self.get_endpoint(target),
+            apikey=self.get_apikey(target, required=auth_required),
+            stack=self.get_stack(target),
+            requirements_file=self.requirements_file,
         )
 
 
+class Target(object):
+
```

Comment:
why a tuple at all?

I tried to keep it working without modifying current code too much but I think returning a tuple is just not good solution